src/main/java/se/sundsvall/esigning/integration/document/DocumentClient.java

```diff
@@ -0,0 +1,34 @@
+package se.sundsvall.esigning.integration.document;
+
+import static se.sundsvall.esigning.integration.document.configuration.DocumentConfiguration.CLIENT_ID;
+
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestPart;
+import org.springframework.web.multipart.MultipartFile;
+
+import generated.se.sundsvall.document.Document;
+import generated.se.sundsvall.document.DocumentUpdateRequest;
+import se.sundsvall.esigning.integration.document.configuration.DocumentConfiguration;
+
+@FeignClient(name = CLIENT_ID, url = "${integration.document.url}", configuration = DocumentConfiguration.class)
+public interface DocumentClient {
+
+	@GetMapping(path = "documents/{registrationNumber}")
+	Document getDocument(@PathVariable("registrationNumber") String registrationNumber);
+
+	@PatchMapping(path = "documents/{registrationNumber}")
+	Document updateDocument(@PathVariable("registrationNumber") String registrationNumber, DocumentUpdateRequest body);
+
+	@GetMapping(path = "documents/{registrationNumber}/files/{documentDataId}")
+	ResponseEntity<ByteArrayResource> getDocumentData(@PathVariable("registrationNumber") String registrationNumber, @PathVariable("documentDataId") String documentDataId);
+
+	@PutMapping(path = "/documents/{registrationNumber}/files", consumes = MediaType.MULTIPART_FORM_DATA_VALUE)
+	Void addFileToDocument(@PathVariable("registrationNumber") String registrationNumber, @RequestPart("document") String document, @RequestPart("documentFile") MultipartFile documentFile);
```

Comment:
Why is document a String?
Isn't DocumentDataCreateRequest `the correct type for the PUT-operation?

Reply:
According to the API the requestpart for the DocumentCreateRequest is to be sent as a String:
```
	@PutMapping(path = "/{registrationNumber}/files", consumes = {MULTIPART_FORM_DATA_VALUE}, produces = {APPLICATION_JSON_VALUE, APPLICATION_PROBLEM_JSON_VALUE})
	@Operation(summary = "Add document file data (or replace existing if filename already exists on the document object)")
	@ApiResponse(responseCode = "200", description = "Successful operation", useReturnTypeSchema = true)
	@ApiResponse(responseCode = "404", description = "Not found", content = @Content(mediaType = APPLICATION_PROBLEM_JSON_VALUE, schema = @Schema(implementation = Problem.class)))
	public ResponseEntity<Void> addOrReplaceFile(
		@Parameter(name = "registrationNumber", description = "Document registration number", example = "2023-2281-1337") @PathVariable("registrationNumber") String registrationNumber,
		@RequestPart("document") @Schema(description = "Document", implementation = DocumentDataCreateRequest.class) String documentDataString,
		@RequestPart(value = "documentFile") MultipartFile documentFile) throws JsonProcessingException {
```
I tried to send it as an json object but the service responded with 400, and that requestPart 'document' was not present in the request on such requests.

I'm not sure, but I guess that there is limitations when using requestPart annotations in the API?

Reply:
![image](https://github.com/Sundsvallskommun/pw-e-signing/assets/46994590/3a95677d-2378-4dd8-9c42-5f1dad2d9c24)

According to the API, "document" should be a DocumentDataCreateRequest

Reply:
Yes but it doesn't work if the object is sent as json in the requestPart. As I said, there are limitations when using requestPart - one is that you can not send json as a requestPart. It has to be converted to string to work (I have googled it and the suggested solution is to serialize the object to a string).

Reply:
I dont understand?

Isn't json sent as string data?
Isn't feign serializing the Java object into a string? (what is sent if it's not a string with json-data?)